src/net/sourceforge/kolmafia/AscensionClass.java

```diff
@@ -47,26 +47,25 @@
     return Arrays.stream(values()).filter(a -> a.getId() > -1).collect(Collectors.toSet());
   }
 
-  public static AscensionClass nameToClass(String name) {
-    if (name.equals("")) {
-      return null;
-    }
+  public static AscensionClass findByPlural(final String plural) {
+    if (plural == null || plural.isEmpty()) return null;
+    return Arrays.stream(values())
+        .filter(a -> a.getPlural().toLowerCase().contains(plural.toLowerCase()))
+        .findFirst()
+        .orElse(null);
+  }
 
-    for (AscensionClass ascensionClass : AscensionClass.values()) {
-      if (ascensionClass.getName().toLowerCase().contains(name.toLowerCase())) {
-        return ascensionClass;
-      }
-    }
-    return null;
+  public static AscensionClass find(String name) {
+    if (name == null || name.equals("")) return null;
+
+    return Arrays.stream(values())
+        .filter(a -> a.getName().toLowerCase().contains(name.toLowerCase()))
```

Comment:
If you bring `name.toLowerCase` out of the stream, you can avoid calculating it each time.

This might be what happens anyway, but I don't think it is.

Same for the other instances of this.